src/index.js

```diff
@@ -1,17 +1,67 @@
 const isDefined = (o) => ![null, undefined].includes(o);
 
-const getFarContext2d = (canvas, { x = 0, y = 0, scale = 1 } = {}) => {
-  const d = { x, y, scale };
+const getFarContext2d = (
+  canvas,
+  { x = 0, y = 0, scale = 1, rotation = { x: 0, y: 0, angle: 0 } } = {}
+) => {
+  if (![0, Math.PI].includes(rotation.angle)) {
+    throw new Error("Only 0 and PI rotation angles are supported");
+  }
+
   const _context = canvas.getContext("2d");
 
+  const d = { x, y: y * Math.cos(rotation.angle), scale, rotation };
+
   const s = {
-    x: (x) => d.scale * (x + d.x),
-    y: (y) => d.scale * (y + d.y),
-    distance: (distance) => distance * d.scale,
+    x: (x) => {
+      // First, translate, then rotate
+      const translatedX = x + d.x;
+      const translatedY = d.y; // y-coordinate remains the same for calculating x
+      // Apply rotation
+      const rotatedX =
+        Math.cos(d.rotation.angle) * (translatedX - d.rotation.x) -
```

Comment:
Math.cos/sin is kinda slow (I'm guessing they don't have a shortcut for PI). But I think it's ok as long as we have much less then 30k objects to render